spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/inbound/HazelcastMessageProducer.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import java.util.Set;
+
+import com.hazelcast.core.DistributedObject;
+
+import reactor.util.StringUtils;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.hazelcast.common.CacheListeningPolicyType;
+import org.springframework.integration.hazelcast.common.HazelcastIntegrationDefinitionValidator;
+import org.springframework.integration.hazelcast.common.HazelcastIntegrationUtils;
+import org.springframework.integration.hazelcast.context.ApplicationContextStartEventHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * This is a base class for Hazelcast Event-Driven Message Producers.
+ * 
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ *
+ */
+public class HazelcastMessageProducer extends MessageProducerSupport implements BeanFactoryPostProcessor, DisposableBean {
+
+	private DistributedObject distributedObject;
+
+	private String cacheEventTypes;
+
+	private CacheListeningPolicyType cacheListeningPolicy;
+
+	@Override
+	protected void onInit() {
+		if (HazelcastIntegrationDefinitionValidator
+				.validateCacheEventByDistributedObject(getDistributedObject(), getCacheEventTypeSet())) {
+			super.onInit();
```

Comment:
The `super.onInit()` as a first line in the method outside of that if.
From other side if our `CacheEventByDistributedObject` isn't valid why just don't throw `IllegalArgumentException`?
I think there might be a reason to revise that `HazelcastIntegrationDefinitionValidator` do not return `true/false`, but throw appropriate exception to prevent our application to start.